app_dart/lib/src/service/branch_service.dart

```diff
@@ -125,8 +125,10 @@ class BranchService {
 
     List<gh.Branch> branches = await github.repositories.listBranches(slug).toList();
     final String devName = await _getDevBranch(github: github, slug: slug, branches: branches);
-    final String betaName = branches.where(((gh.Branch b) => b.commit!.sha == betaSha)).single.name!;
-    final String stableName = branches.where(((gh.Branch b) => b.commit!.sha == stableSha)).single.name!;
+    final String betaName =
+        branches.where(((gh.Branch b) => b.commit!.sha == betaSha && b.name != 'beta' && b.name != 'dev')).single.name!;
```

Comment:
Would it be possible to get two release candidate branches with the same commit as the beta branch?

Reply:
Also if this logic is handling beta should it include the dev check?

Reply:
Right, yes so if two beta branches have the same commit they will both pass the `b.commit!.sha == betaSha` check
And the alias with the name `beta` will be filtered out by the check `b.name != 'beta'`, and the dev branch will be filtered out by the check `b.name != 'dev'`. So I put down `b.commit!.sha == betaSha && b.name != 'beta' && b.name != 'dev'`

Using the test as an example, the following three branch instances
```
      gh.Branch betaBranch = generateBranch(3, name: 'flutter-3.4-candidate.5', sha: '789beta');
      gh.Branch devBranch = generateBranch(4, name: 'dev', sha: '789beta');
      gh.Branch betaAliasBranch = generateBranch(5, name: 'beta', sha: '789beta');
```
will be deduplicated to flutter-3.4-candidate-5

Reply:
I think this is a good example of why we create [release-candidate-branch.version](https://cs.opensource.google/flutter/flutter/+/beta:bin/internal/release-candidate-branch.version).

Can we refactor the logic to use the GitHub files API for finding the beta and stable branches using this version file?

Reply:
Sure, the new commit is now based on file content.